### PR TITLE
Reformat, add more sample tests

### PR DESCRIPTION
**Pull request by fwilhe**
This PR adds a second http client sample spec-file which has the examples exactly as in the documentation.

Those tests us the httpbin.org service so we don't need to work with local ip addresses or hostnames in our documentation.

Also it reformats the code with the repo-wide prettier config.

### CODE REVIEW

File: samples/http-client-examples/http-client-httpbin.spec.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+This spec file includes usage examples for the HTTP client that make use of the httpbin.org service.
+The examples in this file should be exactly the same as on https://sap.github.io/cloud-sdk/docs/js/features/connectivity/generic-http-client
+
+*/
+
+describe('HTTP Client Usage Examples using httpbin.org', () => {
+  it('Show simple HTTP Post without fetchCsrfToken', async () => {
+    const response = await executeHttpRequest(
+      {
+        url: `https://httpbin.org/post`
```
Comment (FrankEssenberger) [inline samples/http-client-examples/http-client-httpbin.spec.ts:18]: [q] is httpbin stable or is a mocked URL the best?

**Review by FrankEssenberger — APPROVED** [review]
LGTM

File: samples/http-client-examples/http-client-httpbin.spec.ts (hunk 1 above)
Comment (fwilhe) [reply samples/http-client-examples/http-client-httpbin.spec.ts:18]: What would a "mocked url" in this case be?

I do have the other spec file which uses our local express-http-server, but I wanted to have tests that are 1:1 the same as on the docs page and don't require a local server.

Sure I can't guarantee how long httpbin will exist, but I know it for years and think it is a quite reliable page for that purpose.

Comment (FrankEssenberger) [reply samples/http-client-examples/http-client-httpbin.spec.ts:18]: discussed in person, that it is ok to use it so the code snippets are also directly usable as copy paste.

**Pull request merged**